modules/tinymce/src/core/main/ts/init/InitContentBody.ts

```diff
@@ -205,6 +205,19 @@ const createParser = function (editor: Editor): DomParser {
     }
   });
 
+  // Convert native media elements to non editable elements
+  parser.addNodeFilter('video,audio,embed,object', (nodes) => {
+    let i = nodes.length;
+    while (i--) {
+      const node = nodes[i];
+      const ce = node.attr('contenteditable');
+      if (ce && !node.attr('data-mce-contenteditable')) {
+        node.attr('data-mce-contenteditable', ce);
+      }
+      node.attr('contenteditable', 'false');
```

Comment:
we don't care to set `data-mce-contenteditable` to match this?

Reply:
`data-mce-contenteditable` is used to preserve the original `contenteditable` attribute if one existed, so we don't want to force it to be false. This is then used in the serializer to restore the original value.

So say this was the HTML input:
```
<video contenteditable="true" src="video.mp4"></video>
```
then when added to the editor, the parser will convert that to:
```
<video contenteditable="false" data-mce-contenteditable="true" src="video.mp4"></video>
```
and then when serializing it'll restore it back to the original.

It's an edge case as not many will use the contenteditable attribute, but I wanted to make sure it was preserved.